// zod/src/v4/classic/external.ts
export * as core from "../core/index";
export * from "./schemas";
export * from "./checks";
export * from "./errors";
export * from "./parse";
export * from "./compat";

// zod-specified
import { config } from "../core/index";
import en from "../locales/en";
config(en());

export type { infer, output, input } from "../core/index";
export {
  globalRegistry,
  type GlobalMeta,
  registry,
  config,
  function,
  $output,
  $input,
  $brand,
  clone,
  regexes,
  treeifyError,
  prettifyError,
  formatError,
  flattenError,
  toJSONSchema,
  TimePrecision,
  NEVER,
} from "../core/index";

export * as locales from "../locales/index";

// iso
// must be exported from top-level
// https://github.com/colinhacks/zod/issues/4491
export { ZodISODateTime, ZodISODate, ZodISOTime, ZodISODuration } from "./iso";
export * as iso from "./iso";

// coerce
export type {
  ZodCoercedString,
  ZodCoercedNumber,
  ZodCoercedBigInt,
  ZodCoercedBoolean,
  ZodCoercedDate,
} from "./coerce";
export * as coerce from "./coerce";
